File: src/packets/PacketC2SDisconnect.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x03
 * Disconnect
 * Informs the server of an official disconnect (Not sure if I need this yet.)
 */
class PacketC2SDisconnect extends PacketC2S {
    parseData() {}

    getID() {
        return PacketC2SDisconnect.ID;
    }
}

PacketC2SDisconnect.ID = 0x03;

module.exports = PacketC2SDisconnect;

File: src/RNetPi.Core/Packets/PacketC2SDisconnect.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x03
/// Disconnect
/// Disconnects from the server
/// </summary>
public class PacketC2SDisconnect : PacketC2S
{
    public const byte ID = 0x03;

    public PacketC2SDisconnect(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        // No data to parse for disconnect packet
    }
}